File: 0932-monotonic-array/0932-monotonic-array.py
class Solution:
    def isMonotonic(self, nums: List[int]) -> bool:
        previous = nums[0]
        monotonic = 0  # 0 uninitialized, 1 increasing, 2 decreasing
        
        for num in nums:
            if num == previous:
                continue
                
            if monotonic:
                if num < previous and monotonic != 2:
                    return False
                elif num > previous and monotonic != 1:
                    return False
            else:
                monotonic = 1 if num > previous else 2
                
            previous = num
        
        return True   

File: 0932-monotonic-array/0932-monotonic-array.cpp
class Solution {
public:
    bool isMonotonic(vector<int>& nums) {
        int previous = nums[0];
        int monotonic = 0; // 0 uninitialized, 1 increasing, 2 decreasing
        for (int num: nums) {
            if (num == previous) continue;
            if (monotonic) {
                if (num < previous && monotonic != 2) {
                    return false;
                } else if (num > previous && monotonic !=1) {
                    return false;
                }
            } else {
                monotonic = (num > previous)? 1 : 2;
            }
            previous = num;
        }
        return true;
    }
};